dbt_coves/tasks/load/base.py

```diff
@@ -22,3 +28,31 @@ def retrieve_all_jsons_from_path(self, path):
                 with open(filepath, "r") as json_file:
                     jsons.append(json.load(json_file))
         return jsons
+
+    def _load_secret_data(self) -> dict:
+        # Contact the manager and retrieve Service Credentials
+        secrets_url = os.getenv("DBT_COVES_SECRETS_URL") or self.get_config_value("secrets_url")
+        secrets_token = os.getenv("DBT_COVES_SECRETS_TOKEN") or self.get_config_value(
+            "secrets_token"
+        )
+        if not (secrets_url and secrets_token):
+            raise LoadException(
+                "[b]secrets_url[/b] and [b]secrets_token[/b] must be provided"
+                "when using a Secrets Manager"
+            )
+        if self.secrets_manager.lower() == "datacoves":
+            secrets_project = self.get_config_value("secrets_project")
+            if not secrets_project:
+                raise LoadException(
+                    "[b]secrets_project[/b] must be provided when using 'datacoves'"
+                    "as your Secrets Manager "
+                )
+            secrets_url = f"{secrets_url}/api/v1/secrets/{secrets_project}"
+            secrets_tags = self.get_config_value("secrets_tags")
+            if secrets_tags:
+                secrets_url += f"?tags={'&'.join(secrets_tags)}"
```

Comment:
You might want to use requests to encode the query string instead of using f-strings. Strictly speaking, anything you put in a url query string parameter should be urlencoded, which requests can do for you. https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls

Reply:
Also, the line seems wrong. It will expand to `?tags=tag1&tag2&tag3` but you probably want `tags=tag1,tag2,tag3` or `&tags=tag1&tags=tag2&tags=tag3`. The `&` separates query string parameters.